### PR TITLE
Adds client behaviour of replicated map to javadoc

### LINKED ISSUE #11707: ReplicatedMap API Doc reflects Member only behaviour
ReplicatedMap Api Doc contains member only behaviours/requirements that client cannot meet  due to its technical difference.

Most of the data providing methods like `Collection<V> values();` promise to return a lazy data set backed by the map where changes to the map might be reflected in the set.

Members returns "a lazy set view of the mappings"  as in api doc,
but client returns "a clone of the mappings" instead. 

As all data exists on all members, such a behaviour easily provided. Whereas client provide simple proxy of a member and always return a clone of data.

The documentation should be review with this in mind.

### PR DESCRIPTION
**Pull request by sancar**
fixes #11707

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/core/ReplicatedMap.java [hunk 1]
```diff
@@ -135,7 +135,8 @@
      *     }
      * </pre>
      * Due to the lazy nature of the returned set, changes to the map (addition,
-     * removal, update) might be reflected in the set.<br/>
+     * removal, update) might be reflected in the collection.<br/>
+     * Changes on the map is <b>NOT</b> reflected in the collection on the <b>CLIENT</b> or vice versa.
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/core/ReplicatedMap.java]: I think it should be plural everywhere: Changes on the map _are_ not reflected...

File: hazelcast/src/main/java/com/hazelcast/core/ReplicatedMap.java [hunk 2]
```diff
@@ -180,6 +181,7 @@
      * </pre>
      * Due to the lazy nature of the returned set, changes to the map (addition,
      * removal, update) might be reflected in the set.<br/>
+     * Changes on the map is <b>NOT</b> reflected in the set on the <b>CLIENT</b> or vice versa.
```
Comment (Serdaro) [inline hazelcast/src/main/java/com/hazelcast/core/ReplicatedMap.java]: Changes on the map are <b>NOT<b> reflected....

File: hazelcast/src/main/java/com/hazelcast/core/ReplicatedMap.java (hunk 1 above)
Comment (Serdaro) [reply hazelcast/src/main/java/com/hazelcast/core/ReplicatedMap.java]: Changes on the map are not.......reflected to the collection....

Comment (sancar) [reply hazelcast/src/main/java/com/hazelcast/core/ReplicatedMap.java]: @Serdaro Fixed.

**Pull request merged**